services/friend_request.service.js

```diff
@@ -1,83 +1,165 @@
+  // console.log(receivingUser)
+  // if (!receivingUser) {
+  //   throw new Error(
+  //     "The user you are trying to send a friend request to doesn't exist"
+  //   );
+  // }
+  let pendingRequestArray = [];
+  let result = await FriendRequest.findAll({
+    where: Sequelize.and({ status: 0 }, { receiverId: userId }),
+  });
+  for (let i in result) {
+    let user = await User.findOne({
+      where: result[i].senderId,
+      include: [
+        { model: UserInterests },
+        {
+          model: PlaceVisited,
+        },
+        {
+          model: PlaceLiving,
+        },
+        {
+          model: PlaceVisiting,
+        },
+      ],
+    });
+    user.dataValues.requestId = result[i].dataValues.id;
+    pendingRequestArray.push(user.dataValues);
+  }
+  return pendingRequestArray;
+};
 
-        let isDuplicateRequest = await requestAlreadySent(current_user_id, receivingUser.id);
-        if (isDuplicateRequest) {
-            throw new Error("Friend request already in progress")
-        }
-        let currentUser = await User.findByPk(current_user_id);
-        let friendRequestObj = {
-            "senderId": currentUser.id,
-            "receiverId": receivingUser.id,
-            "status": 0
-        }
-        let request = await FriendRequest.create(friendRequestObj);
-        let socketFrObj = {
-            "sender": currentUser,
-            "receiver": receivingUser,
-            "friendRequestId": request.id
-        }
-        socket.emit("friend-request", socketFrObj);
-        return request;
-    } catch (e) {
-        throw Error(e)
-    }
-}
+// Load all INCOMING friend requests;
+// Will load all friend which the current user is the RECEIVER of the requests.
+// TODO: Convert to GraphQL versions of INBOUND and OUTBOUND requests.
 
 let loadAllFriendRequests = async (current_user_id) => {
-    let fr = await db.sequelize.query("SELECT fr.id AS fr_id, fr.createdAt as fr_time_Sent, u.id AS sender_id, fr.status, u.username AS sender_username FROM friend_requests AS fr JOIN users u ON u.id = fr.senderId WHERE fr.receiverId = ? AND status = 0", {
-        replacements: [current_user_id],
-        type: Sequelize.QueryTypes.SELECT
-    });
-    return fr;
-}
+  let fr = await db.sequelize.query(
+    'SELECT fr.id AS fr_id, fr."createdAt" as fr_time_Sent, u.id AS "senderId", fr.status, u.username AS sender_username FROM friend_requests AS fr JOIN users u ON u.id = fr."senderId" WHERE fr."receiverId" = ? AND status = 0',
+    { replacements: [current_user_id], type: Sequelize.QueryTypes.SELECT }
+  );
+  return fr;
+};
 
-let acceptFriendRequest = async (friend_request_id) => {
-    try {
-        let updatedRequestId = await FriendRequest.update({ status: 1 }, { where: { id: friend_request_id } });
-        return FriendRequest.findByPk(updatedRequestId[0]);
-    } catch (e) {
-        throw Error(e)
-    }
+let loadCurrentOutboundFriendRequests = async (current_user_id) => {
+  // TODO
```

Comment:
We can likely get rid of this method/the comments around inbound/outbound if we're only looking to support the one approach at the moment.